uint8 ACTION_TYPE_REQUEST_CONTROL = 0
uint8 ACTION_TYPE_RENOUNCE_CONTROL = 1

# request
uint8 action_type
---
uint32 HANDSHAKE_RETURN_CODE_ROBOT_BASE_NOT_ALIVE = 0
uint32 HANDSHAKE_RETURN_CODE_ALREADY_GAINED_CONTROL = 1
uint32 HANDSHAKE_RETURN_CODE_ALREADY_LOST_CONTROL = 2
uint32 HANDSHAKE_RETURN_CODE_CONTROL_ACQUIRED = 3
uint32 HANDSHAKE_RETURN_CODE_CONTROL_REJECTED_ROBOT_BASE_FAULT = 4
uint32 HANDSHAKE_RETURN_CODE_CONTROL_REJECTED_RC_HALT_TRIGGERED = 5
uint32 HANDSHAKE_RETURN_CODE_CONTROL_REJECTED_RC_CONTROL_ACTIVE = 6
uint32 HANDSHAKE_RETURN_CODE_CONTROL_REJECTED_TOKEN_TRANSFER_INTERRUPTED = 7
uint32 HANDSHAKE_RETURN_CODE_CONTROL_HANDED_OVER = 8
uint32 HANDSHAKE_RETURN_CODE_CONTROL_REQUEST_TIMEOUT = 9
uint32 HANDSHAKE_RETURN_CODE_RENOUNCE_REQUEST_TIMEOUT = 10

# response
uint32 result_code